### PR TITLE
Trezor: Enable CLSAG support for Trezor client

### PR DESCRIPTION
**Pull request by ph4r05**
- Enables CLSAG transaction support for Trezor
- Fixes Trezor tests, testing HF \in {12, 13}
- Requires https://github.com/trezor/trezor-firmware/pull/1246
- Testing: https://github.com/ph4r05/monero/releases/tag/v16.0.0.0-tests-u18.04-01
- @SarangNoether for notif

### CODE REVIEW

File: src/device_trezor/trezor/protocol.cpp [hunk 1]
```diff
@@ -1017,14 +1012,23 @@ namespace tx {
       }
     }
 
-    // CLSAG support comes here once it is merged to the Monero
-    m_ct.rv->p.MGs.reserve(m_ct.signatures.size());
-    for(size_t i = 0; i < m_ct.signatures.size(); ++i) {
-      rct::mgSig mg;
-      if (!cn_deserialize(m_ct.signatures[i], mg)) {
-        throw exc::ProtocolException("Cannot deserialize mg[i]");
+    m_ct.rv->p.CLSAGs.reserve(m_ct.signatures.size());
```
Comment (moneromooo-monero) [inline src/device_trezor/trezor/protocol.cpp:1015]: This is technically correct, though it looks like it should be inside the if below, and the reserve for MGs restored in the else.

File: tests/trezor/trezor_tests.cpp [hunk 2]
```diff
@@ -1245,10 +1263,14 @@ void gen_trezor_base::set_hard_fork(uint8_t hf)
   m_top_hard_fork = hf;
   if (hf < 9){
     throw std::runtime_error("Minimal supported Hardfork is 9");
-  } else if (hf == 9){
+  } else if (hf <= 11){
     rct_config({rct::RangeProofPaddedBulletproof, 1});
-  } else {
+  } else if (hf == 12){
     rct_config({rct::RangeProofPaddedBulletproof, 2});
+  } else if (hf == 13){
```
Comment (moneromooo-monero) [inline tests/trezor/trezor_tests.cpp:1270]: There are HF_VERSION_* defines for those if needed.

**Review by moneromooo-monero — APPROVED** [review]
Just a couple minor things which don't prevent this from being merged.

File: tests/trezor/trezor_tests.cpp (hunk 2 above)
Comment (ph4r05) [reply tests/trezor/trezor_tests.cpp]: fixed

File: src/device_trezor/trezor/protocol.cpp (hunk 1 above)
Comment (ph4r05) [reply src/device_trezor/trezor/protocol.cpp]: Aah correct. I've fixed it. Thanks!

**Review by moneromooo-monero — APPROVED** [review]

**Pull request merged**